src/app/page/UserProfile/LinkPhoneAccount.jsx

```diff
@@ -16,23 +16,51 @@ function LinkPhoneAccount({ firebase, auth, data, errorHandler }) {
   const currentUserPhoneNumber = data?.phoneNumber || "";
   const [phoneNumber, updatePhoneNumber] = useState(currentUserPhoneNumber);
   const [successSnackbarOpen, setSuccessSnackbarOpen] = useState(false);
+  const [userHasPhone, setUserHasPhone] = useState(false);
   const classes = useStyles();
 
+  useEffect(() => {
```

Comment:
since `""` evaluates to false we can simply use `currentUserPhoneNumber` as the flag to check and not have to use this extra hook/state.

Reply:
srry I was doing rebase. @qdozaq had to force a push

Reply:
But how to check it when firing the change button? Because by then, it won't be "" anymore

Reply:
`currentUserPhoneNumber` is just coming from the `data` prop. That would only change once we submit the new number. `phoneNumber` is the one thats getting updated on user input

Reply:
@qdozaq u're right. I'll fix it, thanks

Reply:
@qdozaq Done. However @azmainamin said they've tried hard and couldn't make this prop update itself and the component. For that reason there's a window reload call, feel free to dive into it. (I'll go sleep, 11PM here :sleeping:)

Reply:
Thats odd. but as long as it's not breaking anything I guess it's fine for now.
Wonder if something with firebase not triggering a react state change